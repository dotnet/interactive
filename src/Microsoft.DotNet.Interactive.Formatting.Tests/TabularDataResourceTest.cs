// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using System.Text.Json;

using Assent;

using Xunit;

namespace Microsoft.DotNet.Interactive.Formatting.Tests
{
    public class TabularDataResourceTest
    {
        private readonly Configuration _configuration;

        public TabularDataResourceTest()
        {
            _configuration = new Configuration()
                .SetInteractive(Debugger.IsAttached)
                .UsingExtension("json");
        }

        [Fact]
        public void can_create_from_document()
        {
            var doc = JsonDocument.Parse(@"[
{ ""name"": ""mitch"", ""age"": 42, ""salary"":10.0, ""active"":true }
]");
            var data = doc.ToTabularDataResource();
            var formattedData = data.ToDisplayString(TabularDataResourceFormatter.MimeType);

            this.Assent(formattedData, _configuration);
        }
    }
}