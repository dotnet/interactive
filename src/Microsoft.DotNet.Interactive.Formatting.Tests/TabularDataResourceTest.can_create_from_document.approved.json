{
  "profile": "tabular-data-resource",
  "schema": {
    "primaryKey": [],
    "fields": [
      {
        "name": "name",
        "type": "string"
      },
      {
        "name": "age",
        "type": "number"
      },
      {
        "name": "salary",
        "type": "number"
      },
      {
        "name": "active",
        "type": "boolean"
      }
    ]
  },
  "data": [
    {
      "name": "mitch",
      "age": 42,
      "salary": 10,
      "active": true
    }
  ]
}